geode-core/src/distributedTest/java/org/apache/geode/internal/cache/persistence/PersistentRecoveryOrderDUnitTest.java

```diff
@@ -1327,8 +1327,19 @@ public void testRecoverAfterConflict() {
 
     vm0.invoke(() -> {
       // This should work now
-      createReplicateRegion(regionName, getDiskDirs(getVMId()));
-      updateEntry("A", "C");
+      try {
+        createReplicateRegion(regionName, getDiskDirs(getVMId()));
+        updateEntry("A", "C");
+      } catch (CacheClosedException cacheClosedException) {
```

Comment:
If cache gets closed whenever "ConflictingPersistentDataExcpetion" is thrown; will it be a good option to wait for cache to be closed before recreating the replicated-region?
That way it becomes easier to understand the flow/expectation at different blocks of the test.

Reply:
The question is how we can tell the cache is already closed or is in the process of closing, before recreating the region. There is some subtle difference. `GemFireCacheImpl.isClosed()` returns `true`, even when the bug occurs. So we can't depend on the flag `GemFireCacheImpl.isClosed()`. I have also tried to check if the cache is `null`. Not a good idea either. I probably have to do something similar to `cancelInProgress`.

Reply:
isClosed() checks the isClosing flag; which gets set whiel close is in progress.
Checking for null is not a bad idea; to determine the cache is completely closed. 
Other option could be using CacheLifecycleListener...Ex: ModifyColocationIntegrationTest